contracts/FlowCallbackScheduler.cdc

```diff
@@ -850,6 +850,12 @@ access(all) contract FlowCallbackScheduler {
                 for id in callbackIDs.keys {
                     let callback = self.borrowCallback(id: id)!
 
+                    // Callbacks that are already executed are finalized and removed from the queue
```

Comment:
is there a reason we make it so the callback is removed from the queue only after next callback is executed? Meaning what was the thinking to seperate this from finalize being called when executed event is emitted?

Reply:
lol right!